spec/models/game_spec.rb

```diff
@@ -57,4 +57,24 @@
       expect(test).to eq(32)
     end
   end
+  
+  describe 'in_check?' do
+    it 'returns true if game in_check?' do
+      game = FactoryGirl.create(:game)
+      FactoryGirl.create(:piece, x_coordinate: 4, y_coordinate: 4, color: 1, game_id: game.id)
+      FactoryGirl.create(:king, x_coordinate: 4, y_coordinate: 0, game_id: game.id)
+      
+      expect(game.in_check?(0)).to eq(true)
+    end 
+  
```

Comment:
for color you can use 'black' or 'white' instead of integers. Rails does the transition for you